File: recursividade/python/reverse.py
def reverse(lst):
    if len(lst) <= 1:
        return lst

    head = lst[0]
    tail = lst[1:]

    new_list = reverse(tail)
    new_list.append(head)

    return new_list


result = reverse(["azul", "verde", "preto", "rosa"])
print(result)

File: recursividade/javascript/reverse.js
function reverse(list) {
  if (list.length <= 1) {
    return list;
  }

  const head = list[0];
  const tail = list.slice(1);

  const newList = reverse(tail);
  newList.push(head);
  
  return newList;
}

let result = reverse(["azul", "verde", "preto", "rosa"]);
console.log(result);
